src/views/preview/preview.jsx

```diff
@@ -312,6 +324,29 @@ class Preview extends React.Component {
             title: title
         });
     }
+    handleUpdateProjectId (projectId, callback) {
+        // NOTE: this needs more work
+        // NOTE: need to load everything for new project that is loaded in componentDidUpdate
+        this.setState({projectId: projectId}, () => {
+            const parts = window.location.pathname.toLowerCase()
+                .split('/')
+                .filter(Boolean);
+            let newUrl;
+            if (projectId === 0) {
+                newUrl = `/${parts[0]}/editor`;
+                this.props.resetProject();
+            } else {
+                newUrl = `/${parts[0]}/${projectId}/editor`;
+                this.fetchCommunityData();
```

Comment:
In both of these cases, a change to the URL should trigger a change to the `projectId`, which should cause the "init" process to happen in `componentDidUpdate` and trigger these that way.

Reply:
Done, take a look